packages/react-list/Spec.md

```diff
@@ -2,62 +2,375 @@
+
+Both v0 and v8 have components that are explicitly named `List`. However v8 also has `DetailsList` and `GroupedList` which also
+overlap with other UI controls such as trees or tables.
+
+Both libraries share the following features for List:
+
+- Presentational/non-interactive component that serves as layout for a collection of items
+- Selection state for list items
+- Keyboard navigation for list items.
+
+V0 emphasizes layout more, and by default supports a layout with media, main content, and secondary content slots.
+However both libraries allow completely custom content within list items.
+
+Both libraries use shorthand collections through an `items` prop to represent the list items that are rendered
+in the components.
+
+### Selection
+
+v0 and v8 both support selection states for list items. V0 supports standard React control patterns for selecting
+list items, where as v8 uses a non-standard `Selection` class that is passed as reference through props to get the
+current selection state of the list.
+
+```tsx
+// v0
+<List
+  selectable
+  selectedIndex={selectedIndex}
+  onSelectedIndexChange={(e, newProps) => {
+    setState({
+      selectedIndex: newProps.selectedIndex,
+    });
+  }}
+  items={this.items}
+/>;
+
+// v8
+_selection = new Selection({
+  onSelectionChanged: () => {
+    this.setState({
+      selectionDetails: this._getSelectionDetails(),
+    });
+  },
+});
+
+<DetailsList items={items} selectionMode={SelectionMode.multiple} setKey="multiple" selection={_selection} />;
+```
+
+The v8 pattern does not follow standard React design patterns since it relies on either class members
+for React class components, or initialization in a `useState` or `useRef` call in a functional component.
+
+##### Marquee selection
+
+V8 List components support marquee selection (drag a container and select items within the container) through the
+addition of a `MarqueeSelection` component that wraps the list component. The non-standard selection API is used to power
+selection by sharing the same reference to the `Selection` object.
+
+```tsx
+<MarqueeSelection selection={this._selection}>
+  <DetailsList selection={this._selection} />
+</MarqueeSelection>
+```
+
+#### Appearance
+
+Both components have similar philosophies for displaying a vertical list of items. V0 has a stronger layout
+implication that v8, but is still fully customizable if needed.
+
+Grid layout is a scenario v8 has documented that is not supported by the v0 `List` at all.
+
+Compact layout is also possible with `DetailsList`, which reduces the extra spacing between list items and removes
+padding with the boundary of the list.
+
+v8 selection scenarios for `List` components will use a checkbox for each item, whereas v0 will use background color
+changes to indicate selection.
+
+#### Virtualization
```

Comment:
Which virtualization approach will you propose for v9 lists?

Reply:
I think as a part of the development we need to test against several

Reply:
I'd suggest to have virtualization decoupled with composability in mind + not throwing lot of unnecessary JS paylod to consumer, rather than creating some hard coded solution.